### DISCUSSION TITLE
Time between dispatched and started workflows

### DISCUSSION THREAD

**Discussion by tsoG2000** [elsa-workflows/elsa-core#1622] [Q&A] [answered]
When we did performance test, a lot of workflows took 10-15 minutes to start after being dispatched is there some configuration to speed things up?

We did some configuring with number of workers as exampled here:

https://github.com/elsa-workflows/elsa-core/discussions/1308 

but without much benefit.

**Comment by sfmskywalker** [comment, marked as answer]
I presume that the performance test created a lot of workflows that you then dispatched. When dispatching large numbers of workflows, you are limited by the number of workers that can process the workflow queue.

For example, if you dispatched 30.000 workflows but the workers combined can only process 100 per second, it will take 300 seconds to process all workflows. It will then take 5 minutes before the last workflows can execute.

To increase throughput, you need to add more workers and [move heavy lifting out of the workflow](https://github.com/elsa-workflows/elsa-core/discussions/1618#discussioncomment-1464985).

But, even doing that, there are only so many workers that can execute at the same time, depending on your server's hardware & number of processors. Which means that you might need to spin up multiple server nodes in a cluster. Each node will then have its own set of workers, all of them now being able to feed off the workflow queue in parallel.
